Bottom: 78e9a33ca19576a7097e8922b643b0b3699e683c
Top:    0432b3951213604f38935ad281bb36acabcefed3
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-05 12:36:01 +0000

Add void *userdata member to the POLYAREA object

This is intended to allow client code operating on polygons to assign
arbitrary additional data to the polygon contours.

The initial target use-case is in 3D export code, where it is helpful to
have temporary links between POLYAREA contours being extruded and the 3D
object structures being generated. Allowing this tagging directly avoids
the need to maintain a hash-mapping or similar.



---

diff --git a/src/polyarea.h b/src/polyarea.h
index 76b724a..bcb2152 100644
--- a/src/polyarea.h
+++ b/src/polyarea.h
@@ -159,6 +159,7 @@ struct POLYAREA
     PLINE *contours;
     rtree_t *contour_tree;
     POLYPARENTAGE parentage;
+    void *user_data; /* To be used at will by the owner of this polygon */
 };
 
 void poly_dump (POLYAREA *p);
diff --git a/src/polygon1.c b/src/polygon1.c
index 8b99dac..c5b2d71 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -3902,6 +3902,7 @@ poly_Init (POLYAREA * p)
   p->contours = NULL;
   p->contour_tree = r_create_tree (NULL, 0, 0);
   p->parentage = no_parentage;
+  p->user_data = NULL;
 }
 
 POLYAREA *
